src/main/webapp/app/course/manage/course-management-exercises.component.ts

```diff
@@ -28,14 +36,47 @@ export class CourseManagementExercisesComponent implements OnInit {
     ngOnInit(): void {
         this.courseId = Number(this.route.parent!.snapshot.paramMap.get('courseId'));
         this.courseService.find(this.courseId).subscribe((courseResponse) => (this.course = courseResponse.body!));
+        this.exerciseFilter = new ExerciseFilter('');
     }
 
     /**
-     * Sets the programming exercise count. Required to pass a callback to the
+     * Sets the (filtered) programming exercise count. Required to pass a callback to the
      * overrideProgrammingExerciseCard extension since extensions don't support @Output
      * @param count to set the programmingExerciseCount to
      */
     setProgrammingExerciseCount(count: number) {
         this.programmingExercisesCount = count;
     }
+    setFilteredProgrammingExerciseCount(count: number) {
+        this.filteredProgrammingExercisesCount = count;
+    }
+
+    /**
+     * Toggles the search bar
+     */
+    toggleSearch() {
+        this.showSearch = !this.showSearch;
+        // Reset the filter when the search bar is closed
+        if (!this.showSearch) {
+            this.exerciseFilter = new ExerciseFilter();
+        }
+    }
+
+    getExerciseCount(): number {
+        return this.quizExercisesCount + this.programmingExercisesCount + this.modelingExercisesCount + this.fileUploadExercisesCount + this.textExercisesCount;
+    }
+
+    getFilteredExerciseCount(): number {
+        return (
+            this.filteredProgrammingExercisesCount +
+            this.filteredQuizExercisesCount +
+            this.filteredModelingExercisesCount +
+            this.filteredTextExercisesCount +
+            this.filteredFileUploadExercisesCount
+        );
+    }
+
+    shouldHideExerciseCard(type: string): boolean {
+        return !['all', type].includes(this.exerciseFilter.exerciseTypeSearch);
```

Comment:
Could you replace all these functions with ones that set corresponding variables? On update, you could then call these functions one time and the attributes get updated.
(Coding Guidelines)